homeassistant/components/yamaha_musiccast/__init__.py

```diff
@@ -1 +1,171 @@
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from ...helpers.entity import DeviceInfo, Entity
+from .const import BRAND, DOMAIN
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+
+PLATFORMS = ["media_player"]
+
+_LOGGER = logging.getLogger(__name__)
+SCAN_INTERVAL = timedelta(seconds=60)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the MusicCast component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up MusicCast from a config entry."""
+
+    client = MusicCastDevice(
+        entry.data[CONF_HOST], async_get_clientsession(hass), hass.loop
+    )
+    coordinator = MusicCastDataUpdateCoordinator(hass, client=client)
+    await coordinator.async_refresh()
+
+    if not coordinator.last_update_success:
+        raise ConfigEntryNotReady
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = coordinator
+
+    @service.verify_domain_control(hass, DOMAIN)
+    async def async_group_service_handle(service_call: ServiceCall):
+        """Handle services."""
+        entity_ids = service_call.data.get("entity_id", [])
+        if not entity_ids:
+            return
+
+        all_entities = []
+        for coord in hass.data[DOMAIN].values():
+            all_entities += coord.entities
+
+    for component in PLATFORMS:
+        coordinator.platforms.append(component)
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    entry.add_update_listener(async_reload_entry)
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+async def async_reload_entry(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Reload config entry."""
+    await async_unload_entry(hass, entry)
+    await async_setup_entry(hass, entry)
+
+
+class MusicCastDataUpdateCoordinator(DataUpdateCoordinator[MusicCastData]):
+    """Class to manage fetching data from the API."""
+
+    def __init__(self, hass: HomeAssistant, client: MusicCastDevice) -> None:
+        """Initialize."""
+        self.musiccast = client
+        self.platforms: list[str] = []
+        self.entities: list[Entity] = []
+
+        super().__init__(hass, _LOGGER, name=DOMAIN, update_interval=SCAN_INTERVAL)
+
+    async def _async_update_data(self) -> MusicCastData:
+        """Update data via library."""
+        try:
+            await self.musiccast.fetch()
+            return self.musiccast.data
+        except Exception as exception:
+            raise UpdateFailed() from exception
+
```

Comment:
Please only wrap the line that can raise in the try... except block.
```suggestion
        except Exception as exception:
            raise UpdateFailed() from exception
        return self.musiccast.data

```